lib/cli-engine.js

```diff
@@ -97,7 +98,9 @@ var defaultOptions = {
         cacheFile: ".eslintcache",
         fix: false,
         allowInlineConfig: true,
-        cwd: process.cwd()
+        warnIgnoredByDefault: true,
+        cwd: process.cwd(),
+        onOsCompatibilityIssue: logging.info
```

Comment:
The `onOsCompatibilityIssue` will be removed. `warnIgnoredByDefault` is used to suppress warnings when ignoring files looked up by glob pattern, but not when passed the names explicitly. If I pass in a file to lint explicitly, I would expect a warning when this file is ignored for some reason.  We could get rid of this when explicitly specified files override ignore patterns.